src/pip/_internal/download.py

```diff
@@ -222,6 +354,21 @@ def warn_on_401(self, resp, **kwargs):
             logger.warning('401 Error, Credentials not correct for %s',
                            resp.request.url)
 
+    def save_credentials(self, resp, **kwargs):
+        # save stored credentials if they worked
```

Comment:
This looks like it wants to be a docstring rather than comment.

Reply:
So do the ones in the methods above. I assume since these are callbacks rather than APIs that omitting the docstring was deliberate. (As above, should I update the adjacent methods that look the same?)

Reply:
Same with my previous comment. I wouldn't want to make you fix the others, but I'd be okay with you fixing them.